File: languages/java/src/main/java/com/bloock/sdk/entity/identity/DatetimeAttributeDescriptor.java
package com.bloock.sdk.entity.identity;

import com.bloock.sdk.bridge.proto.BloockIdentityEntities;

/**
 * Represents a descriptor for an attribute with a datetime value.
 */
public class DatetimeAttributeDescriptor extends AttributeDescriptor {
  /**
   * Constructs an DatetimeAttributeDescriptor object with the specified
   * parameters.
   * 
   * @param displayName
   * @param technicalName
   * @param description
   * @param required
   */
  public DatetimeAttributeDescriptor(
      String displayName, String technicalName, String description, Boolean required) {
    super(displayName, technicalName, description, required);
  }

  public static DatetimeAttributeDescriptor fromProto(
      BloockIdentityEntities.DateTimeAttributeDefinition res) {
    return new DatetimeAttributeDescriptor(
        res.getDisplayName(), res.getId(), res.getDescription(), res.getRequired());
  }

  public BloockIdentityEntities.DateTimeAttributeDefinition toProto() {
    return BloockIdentityEntities.DateTimeAttributeDefinition.newBuilder()
        .setDisplayName(this.displayName)
        .setId(this.technicalName)
        .setDescription(this.description)
        .setRequired(this.required)
        .build();
  }
}

File: languages/go/entity/identityV2/datetime_attribute_descriptor.go
package identityV2

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// DatetimeAttributeDescriptor represents a descriptor for an attribute with a datetime value.
type DatetimeAttributeDescriptor struct {
	AttributeDescriptor
}

// NewDatetimeAttributeDescriptor creates a new DatetimeAttributeDescriptor instance with the provided details.
func NewDatetimeAttributeDescriptor(name string, id string, description string, required bool) DatetimeAttributeDescriptor {
	a := AttributeDescriptor{
		DisplayName: name,
		Id:          id,
		Description: description,
		Required:    required,
	}
	return DatetimeAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func NewDatetimeAttributeDescriptorFromProto(s *proto.DateTimeAttributeDefinitionV2) DatetimeAttributeDescriptor {
	if s == nil {
		return DatetimeAttributeDescriptor{}
	}
	a := AttributeDescriptor{
		DisplayName: s.GetId(),
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
	return DatetimeAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func (s DatetimeAttributeDescriptor) ToProto() *proto.DateTimeAttributeDefinitionV2 {
	return &proto.DateTimeAttributeDefinitionV2{
		DisplayName: s.DisplayName,
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
}
